Samples/Flix/src/main/java/dev/testify/samples/flix/presentation/moviedetails/viewmodel/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package dev.testify.samples.flix.presentation.moviedetails.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dev.testify.samples.flix.application.foundation.ui.state.*
+import dev.testify.samples.flix.domain.model.MovieDetailsDomainModel
+import dev.testify.samples.flix.domain.usecase.LoadMovieDetailsUseCase
+import dev.testify.samples.flix.presentation.moviedetails.model.MovieDetailsPresentationModel
+import dev.testify.samples.flix.presentation.moviedetails.model.mapper.MovieDetailsDomainModelToPresentationMapper
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+sealed class MovieDetailsViewState : ViewState() {
+    data class LoadedMovieDetailsViewState(val presentationModel: MovieDetailsPresentationModel) : MovieDetailsViewState()
+}
+
+class MovieDetailsViewModel(
+    private val args: Args,
+    private val loadMovieDetailsUseCase: LoadMovieDetailsUseCase,
+    private val movieDetailsPresentationMapper: MovieDetailsDomainModelToPresentationMapper
+) : ViewModel() {
+
+    companion object {
+        private val LOG_TAG = MovieDetailsViewModel::class.simpleName
+    }
+
+    data class Args(val movieId: Int)
+
+    val screenState: StateFlow<ScreenState>
+        get() = _screenState
+
+    private val _screenState: MutableStateFlow<ScreenState> = MutableStateFlow(UnsetScreenState)
```

Comment:
Might be worthwhile to make a `BaseViewModel` that has this and related helper functions in it so that you don't have to repeat the implementation across multiple screens.